content/6_Advanced/LineContainer.mdx

```diff
@@ -30,13 +30,166 @@ frequency: 1
+
+So, as an overview of the algorithm:
+* Sort the set of half-planes by angle; it takes $\mathcal{O}(NlogN)$ time.
+* Iterate over the set. For each, perform incrementation and pop either from the front or back of the double-ended queue as necessary; it takes linear time since every half-plane can only be added or removed once.
+* Once this has been completed, the convex polygon that results from the intersection can be obtained by computing the intersection points of adjacent half-planes in the double-ended queue. It will take linear time. This results in our final complexity as $\mathcal{O}(NlogN)$. In a special case where the half-planes are pre-sorted, this algorithm could be completed in $\mathcal{O}(N)$. However, for the majority of cases, we will be looking at $\mathcal{O}(N \log N)$, most of which is coming from the sorting by angle.
+
+## Implementation
+
+**Time Complexity: $\mathcal{O}(N \log N)$**
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <algorithm>
+#include <climits>
+#include <deque>
+#include <vector>
+
+const long double eps =
+    1e-9;                  /** @Small epsilon for floating-point comparison */
+const int inf = INT32_MAX; /** @Infinite value for bounding box */
+
+struct Point {
+	long double x, y;
+	explicit Point(long double x = 0, long double y = 0) : x(x), y(y) {}
+
+	/** @comment Operator overloads for Point operations */
+};
+
+struct Halfplane {
+	Point p, pq;       /** @Point and direction vector of half-plane */
+	long double angle; /** @Angle of the direction vector */
+
+	Halfplane() {}
+	Halfplane(const Point &a, const Point &b) : p(a), pq(b - a) {
+		angle = atan2l(pq.y, pq.x); /** @Compute angle of direction vector */
+	}
+
+	/** @Check if a point r is outside the half-plane */
+	bool out(const Point &r) { return cross(pq, r - p) < -eps; }
+
+	/** @Operators for sorting half-planes based on angle */
+};
+
+/** @Compute the intersection of a set of half-planes */
+std::vector<Point> halfplane_intersect(std::vector<Halfplane> &halfplanes) {
+	Point box[4] = {Point(inf, inf), Point(-inf, inf), Point(-inf, -inf),
+	                Point(inf, -inf)};
+	/** @Define a bounding box as 4 half-planes */
+
+	/** @Add bounding box halfplanes to the input set */
+	for (int i = 0; i < 4; i++) {
+		Halfplane aux(box[i], box[(i + 1) % 4]);
+		halfplanes.push_back(aux);
+	}
+
+	/** @Sort half-planes by angle (counter-clockwise) */
+	std::sort(halfplanes.begin(), halfplanes.end());
+
+	/** @Remove duplicates (half-planes with same direction) */
```

Comment:
uuuhhh idt these are valid doc comment things lol
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html
see here for valid tags

Reply:
oh ok lol just saw. i'll fix them.